simphony/io/tests/test_serialisation.py

```diff
@@ -0,0 +1,147 @@
+"""
+    Testing module for CUDS serialization functions.
+"""
+import unittest
+import os
+import shutil
+from contextlib import closing
+import tempfile
+import numpy
+import string
+import random
+from importlib import import_module
+
+from simphony.cuds.meta.cuds_component import CUDSComponent
+from simphony.cuds.model import CUDS
+from simphony.core.keywords import KEYWORDS
+from simphony.core.cuba import CUBA
+from simphony.io.serialisation import save_CUDS, load_CUDS
+from simphony.cuds.meta.validation import to_camel_case
+
+
+class TestSerialisation(unittest.TestCase):
+    """Tests for CUDS Yaml serialisation functions."""
+    def setUp(self):
+        self.temp_dir = tempfile.mkdtemp()
+
+    def tearDown(self):
+        shutil.rmtree(self.temp_dir)
+
+    def test_save_CUDS_name(self):
+        name = 'somename'
+        filename = os.path.join(self.temp_dir, 'test_named.yml')
+        C = CUDS(name=name)
+        with closing(open(filename, 'w')) as handle:
+            save_CUDS(handle, C)
+
+        with closing(open(filename, 'r')) as handle:
+            CC = load_CUDS(handle)
+        self.assertEqual(CC.name, name)
+
+    def test_save_CUDS_description(self):
+        description = 'some very long description'
+        filename = os.path.join(self.temp_dir, 'test_description.yml')
+        C = CUDS(description=description)
+        with closing(open(filename, 'w')) as handle:
+            save_CUDS(handle, C)
+
+        with closing(open(filename, 'r')) as handle:
+            CC = load_CUDS(handle)
+        self.assertEqual(CC.description, description)
+
+    def test_save_CUDS_empty(self):
+        filename = os.path.join(self.temp_dir, 'test_empty.yml')
+        C = CUDS(name='empty', description='just an empty model')
+        with closing(open(filename, 'w')) as handle:
+            save_CUDS(handle, C)
+
+        with closing(open(filename, 'r')) as handle:
+            CC = load_CUDS(handle)
+
+        self.assertEqual(CC.name, C.name)
+        self.assertEqual(CC.description, C.description)
+
+        for item in CC.iter(CUDSComponent):
+            self.assertEqual(item, None)
+
+    def test_save_CUDS_full(self):
+        filename = os.path.join(self.temp_dir, 'test_full.yml')
+        C = CUDS(name='full', description='fully randomized model')
+
+        for key in KEYWORDS.keys():
+            if key not in ['VERSION']:
```

Comment:
Version() needs four mandatory arguments and therefore we can't use ```comp_inst = comp_class(None,None)``` (see serialisation.py L242) to instantiate it. I could try to modify the L242 so that it would check there how many supported parameters the class has and instantiate it with corresponding number of Nones.